test/mangata-instance.test.ts

```diff
@@ -1,64 +1,29 @@
-  const getNodeVersion = (Mangata.prototype.getNodeVersion = jest.fn())
-
-  it('should be always the same instance', async () => {
-    const m1 = Mangata.getInstance('ws://127.0.0.1:9944')
-    expect(m1).toEqual(m)
-  })
+const uri = process.env.API_URL ? process.env.API_URL : 'ws://127.0.0.1:9944'
+const m = Mangata.getInstance(uri)
 
+describe('test singleton instance', () => {
   it('should retrive chain name when calling getChain method', async () => {
-    getChain.mockResolvedValue('Mangata')
-    expect(await getChain()).toEqual('Mangata')
-    expect(getChain.mock.calls.length).toBe(1)
-    getChain.mockReset()
-  })
-
-  it('should retrive chain name always when calling getChain method', async () => {
-    getChain.mockResolvedValue('Mangata')
-    expect(await getChain()).toEqual('Mangata')
-    expect(await getChain()).toEqual('Mangata')
-    expect(await getChain()).toEqual('Mangata')
-    expect(getChain.mock.calls.length).toBe(3)
-    getChain.mockReset()
-  })
-
-  it('should retrive node name when calling getNodeName method', async () => {
-    getNodeName.mockResolvedValue('Mangata Node Name')
-    expect(await getNodeName()).toEqual('Mangata Node Name')
-    expect(getNodeName.mock.calls.length).toBe(1)
-    getNodeName.mockReset()
+    const chain = await m.getChain()
+    expect(chain).toEqual('Development')
   })
 
-  it('should retrive node name always when calling getNodeName method', async () => {
-    getNodeName.mockResolvedValue('Mangata Node Name')
-    expect(await getNodeName()).toEqual('Mangata Node Name')
-    expect(await getNodeName()).toEqual('Mangata Node Name')
-    expect(await getNodeName()).toEqual('Mangata Node Name')
-    expect(getNodeName.mock.calls.length).toBe(3)
-    getNodeName.mockReset()
+  it('should match version 2.0.0 node version when calling getNodeVersion method', async () => {
+    const version = '2.0.0'
+    const nodeVersion = await m.getNodeVersion()
+    expect(nodeVersion).toMatch(new RegExp(`^${version}?`))
   })
 
-  it('should retrive node version when calling getNodeVersion method', async () => {
-    getNodeVersion.mockResolvedValue('0.9.6')
-    expect(await getNodeVersion()).toEqual('0.9.6')
-    expect(getNodeVersion.mock.calls.length).toBe(1)
-    getNodeVersion.mockReset()
+  it('should match version 2.0.0 node version when calling getNodeVersion method', async () => {
+    const name = 'Substrate'
+    const nodeName = await m.getNodeName()
+    expect(nodeName).toMatch(new RegExp(`^${name}?`))
   })
+})
 
-  it('should retrive node version always when calling getNodeVersion method', async () => {
-    getNodeVersion.mockResolvedValue('0.9.6')
-    expect(await getNodeVersion()).toEqual('0.9.6')
-    expect(await getNodeVersion()).toEqual('0.9.6')
-    expect(await getNodeVersion()).toEqual('0.9.6')
-    expect(getNodeVersion.mock.calls.length).toBe(3)
-    getNodeVersion.mockReset()
-  })
+afterAll(async () => {
+  const api = await m.getApi()
+  api.disconnect()
```

Comment:
in future, please wrap `disconnect` as well

Reply:
yes I will include this into class